relay-server/src/actors/upstream.rs

```diff
@@ -1378,7 +1412,10 @@ impl UpstreamBroker {
                     entry.retries += 1;
                     action_tx.send(Action::Retry(entry)).ok();
                 }
-                _ => entry.request.respond(result).await,
+                _ => {
+                    entry.request.respond(result).await;
+                    action_tx.send(Action::SuccessfulRequest).ok();
```

Comment:
shouldn't this only send if the `RequestOutcome` is of variant `Received` ? here it will be sent if it's `Dropped` and the entry is not marked as `retry`

theres essentially 3 arms here

`RequestOutcome::Dropped && entry.request.retry() == true`,
`RequestOutcome::Dropped && entry.request.retry() == false`,
`RequestOutcome::Received`,

I think we should only reset the retry-backoff on the final arm